### ISSUE TITLE
If you follow someone before they verify a link from a different server, it doesn't show up as verified.

### ISSUE THREAD

**Issue by donkoch** [mastodon/mastodon#22449]
### Steps to reproduce the problem

Guessing this is the sitution:
1. From server X, follow someone on server Y before they verified a link.
2. Said someone then verifies their account via a link.
3. It shows up verified on Y, but never on X.

### Expected behaviour

It should show up verified on all servers.

### Actual behaviour

It only shows up verified on servers that see the acccount after it has already been verified on the original server.

### Detailed description

I'm seeing accounts verified on their server, but never on mine.

### Specifications

Mastodon 4.0.2 on both servers.

**Comment by ineffyble** [comment]
If you search for them from Server X in the search bar, does it then show as verified?

**Comment by donkoch** [comment]
I don't think so, but the one case I knew of has since been fixed by some means (_e.g._ one can fix it by deleting the link and recreating it) and I haven't found another one.

**Comment by mckinnon** [comment]
I was just discussing this bug with someone hosted on Mastodon.social (v.v4.0.2). I see his verified links but he doesn't.
